### PR TITLE
Move to es6

### PR DESCRIPTION
**Pull request by iherman**
Changed the code so that:

- using es6 features
- removed the need for jQuery, moment, or underscore

### CODE REVIEW

**Comment by iherman** [comment]
@plehegar moving to a respec-style config needs more time; I would prefer to have this version done. In theory, it should be backward compatible, ie, all current files should work even if they would unnecessarily load jQuery and try to load moment and underscore. (Of course, these links should be removed).

I will have to think about an update without harming deployed errata.

File: assets/errata.js [hunk 1]
```diff
@@ -13,115 +13,160 @@ a new comment on the issue may be added beginning with the word "Summary".
...
+*/
 
-TODO: This code should be re-written...
+/**
+ * Fetch a JSON file.
+ * 
+ * @async
+ * @param {string} resource_url 
+ */
+async function fetch_json(resource_url) {
```
Comment (plehegar) [inline assets/errata.js]: How about using [json()](https://fetch.spec.whatwg.org/#dom-body-json) instead
`async function fetch_json(resource_url) {
  return  fetch(resource_url).then(response => response.json());
}`

File: assets/errata.js [hunk 2]
```diff
@@ -13,115 +13,160 @@ a new comment on the issue may be added beginning with the word "Summary".
...
-        div.append("<h3>\"" + issue.title + "\"</h3>");
-
-        let state = 'Open';
-        let state_class = 'state_open';
-        if (issue.state !== 'open') {
-            state = 'Closed';
-            state_class = 'state_closed';
+    // See if a summary has been added to the comment.
+    let summary = undefined;
+    comments.forEach((comment) => {
```
Comment (plehegar) [inline assets/errata.js]:
```suggestion
    comments.forEach(comment => {
```

File: assets/errata.js [hunk 3]
```diff
@@ -13,115 +13,160 @@ a new comment on the issue may be added beginning with the word "Summary".
...
-        div.append("<p><span class='what'><a href='" + issue.html_url + "'>Initial description:</a></span> " + issue.body + "</p>");
-
-        // See if a summary has been added to the comment.
-        var summary = undefined;
-        _.each(comments, function(comment) {
-            if( comment.body.search("^Summary:") !== -1 ) {
-                summary = comment;
-            }
-        })
+    if( summary !== undefined ) {
```
Comment (plehegar) [inline assets/errata.js]:
```suggestion
    if( summary ) {
```

File: assets/errata.js [hunk 4]
```diff
@@ -13,115 +13,160 @@ a new comment on the issue may be added beginning with the word "Summary".
...
+ * 
+ * - Find the section corresponding to a document via a match of a label in the HTML section element; if not found, display the issue in the 'nolabel' section
+ * - Find, within the target, the subsection corresponding to an Editorial issue (if so marked) or the substantial one.
+ * 
+ * @param {object} issue - the issue structure as returned by the Github API
+ * @param {object[]} comments - array of comment objects, as returned by the Github API
+ */
+function render_issue(issue, comments) {
+    // Get the labels as an array of strings
+    const labels = issue.labels.map((obj) => obj.name);
```
Comment (plehegar) [inline assets/errata.js]:
```suggestion
    const labels = issue.labels.map(obj => obj.name);
```

File: assets/errata.js [hunk 5]
```diff
@@ -13,115 +13,160 @@ a new comment on the issue may be added beginning with the word "Summary".
...
+    const get_subsection = (node, lbls) => lbls.includes('Editorial') ? node.querySelector('section:last-of-type') : node.querySelector('section:first-of-type');
 
-        if( summary !== undefined ) {
-            summary_text = summary.body.substr("Summary:".length)
-            div.append("<p><span class='what'><a href='" + summary.html_url + "'>Erratum summary:</a></span> " + summary_text + "</p>");
+    let displayed = false;
+    const sections = document.querySelectorAll('main > section');
+
+    // first looking for sections corresponding to the main errata labels
+    sections.forEach((section) => {
```
Comment (plehegar) [inline assets/errata.js]:
```suggestion
    sections.forEach(section => {
```

File: assets/errata.js [hunk 6]
```diff
@@ -13,115 +13,160 @@ a new comment on the issue may be added beginning with the word "Summary".
...
-                if( _.include(labels, "Editorial") ) {
-                    subsect = $(this).children("section:last-of-type")
-                } else {
-                    subsect = $(this).children("section:first-of-type")
-                }
-            display_issue(subsect, issue, comments, labels)
+    // As a fallback, look for the section marked for 'nolabel' entries.
+    if (!displayed) {
+        // looking for the 'nolabel' data attribute
+        sections.forEach((section) => {
```
Comment (plehegar) [inline assets/errata.js]:
```suggestion
        sections.forEach(section => {
```

File: assets/errata.js (hunk 1 above)
Comment (iherman) [reply assets/errata.js]: Oops, I missed this one; did not realize it has this built-in feature. Neat.

File: assets/errata.js (hunk 2 above)
Comment (iherman) [reply assets/errata.js]: That is a question of style... I usually prefer to keep the arrow function's arguments in brackets, even if there is only one of them.

File: assets/errata.js (hunk 4 above)
Comment (iherman) [reply assets/errata.js]: See above

File: assets/errata.js (hunk 5 above)
Comment (iherman) [reply assets/errata.js]: See above.

File: assets/errata.js (hunk 6 above)
Comment (iherman) [reply assets/errata.js]: See above

File: assets/errata.js (hunk 3 above)
Comment (iherman) [reply assets/errata.js]: yep... I have got bitten by this type of shorthand in Javascript before, hence my instinctive reaction of spelling things out. But I guess in this case it is all right.

**Pull request merged**